linera-views/src/scylla_db.rs

```diff
@@ -186,75 +187,54 @@ impl ScyllaDbClientInternal {
-                let query = format!("DELETE FROM kv.{} WHERE k >= ?", table_name);
-                session.query(query, values).await?;
-            }
-            Some(upper_bound) => {
-                let values = (key_prefix, upper_bound);
-                let query = format!(
-                    "DELETE FROM kv.{} WHERE dummy = 0 AND k >= ? AND k < ?",
-                    table_name
-                );
-                session.query(query, values).await?;
-            }
-        }
-        Ok(())
-    }
-
     async fn write_batch_internal(
         client: &ScyllaDbClientPair,
         batch: Batch,
     ) -> Result<(), ScyllaDbContextError> {
+        let session = &client.0;
+        let table_name = &client.1;
+        let mut batch_query =
+            scylla::statement::batch::Batch::new(scylla::frame::request::batch::BatchType::Logged);
+        let mut batch_values = Vec::new();
         for ent in batch.operations {
-            match ent {
+            let (query, values) = match ent {
                 WriteOperation::Put { key, value } => {
-                    Self::insert_key_value_internal(client, key, value).await?;
+                    let query = format!(
+                        "INSERT INTO kv.{} (dummy, k, v) VALUES (0, ?, ?)",
+                        table_name
+                    );
+                    let values = vec![key, value];
+                    (query, values)
                 }
                 WriteOperation::Delete { key } => {
-                    Self::delete_key_internal(client, key).await?;
+                    let query = format!("DELETE FROM kv.{} WHERE dummy = 0 AND k = ?", table_name);
+                    let values = vec![key];
+                    (query, values)
                 }
                 WriteOperation::DeletePrefix { key_prefix } => {
-                    Self::delete_key_prefix_internal(client, key_prefix).await?;
+                    match get_upper_bound_option(&key_prefix) {
+                        None => {
+                            let values = vec![key_prefix];
+                            let query =
+                                format!("DELETE FROM kv.{} WHERE dummy = 0 AND k >= ?", table_name);
+                            (query, values)
+                        }
+                        Some(upper_bound) => {
+                            let values = vec![key_prefix, upper_bound];
+                            let query = format!(
+                                "DELETE FROM kv.{} WHERE dummy = 0 AND k >= ? AND k < ?",
+                                table_name
+                            );
+                            (query, values)
+                        }
+                    }
                 }
-            }
+            };
+            batch_values.push(values);
+            let query = Query::new(query);
+            batch_query.append_statement(query);
         }
+        session.batch(&batch_query, batch_values).await?;
```

Comment:
(Just for my own understanding: This is the main point of the PR, is it? That we make one batch call to the session for the whole batch, instead of one query per entry, so that it's an atomic operation?)

Reply:
Yes.